vscode-smart-contracts/src/commands.ts

```diff
@@ -15,10 +19,155 @@ export async function version() {
     const version = await cargoConcordium.version();
     vscode.window.showInformationMessage(version);
   } catch (error) {
-    if (error instanceof ConfigError) {
+    if (error instanceof config.ConfigError) {
       vscode.window.showErrorMessage(error.message);
     } else {
-      throw error;
+      vscode.window.showErrorMessage("Unexpected error: " + error);
+    }
+  }
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * Printing the schema in base64 as part of stdout.
+ */
+export function build(editor: vscode.TextEditor) {
+  return buildWorker(editor, "stdout");
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * With no schema generation.
+ */
+export function buildSkipSchema(editor: vscode.TextEditor) {
+  return buildWorker(editor, "skip");
+}
+
+/**
+ * Run 'cargo-concordium build' using the directory of the currently focused editor.
+ * Embedding the schema into the resulting smart contract.
+ */
+export function buildEmbedSchema(editor: vscode.TextEditor) {
+  return buildWorker(editor, "embed");
+}
+
+/**
+ * Type representing the different settings for schema generation during cargo-concordium build.
+ */
+type SchemaSettings = "skip" | "embed" | "stdout";
+
+/**
+ * Internal worker for running 'cargo-concordium build' using the directory of the currently focused editor.
+ * Takes the schema setting as an argument.
+ */
+async function buildWorker(
+  editor: vscode.TextEditor,
+  schemaSettings: SchemaSettings
+) {
+  if (!(await haveWasmTargetInstalled())) {
+    const response = await vscode.window.showInformationMessage(
+      "The needed wasm32-unknown-unknown rust target seems to be missing. Should it be installed?",
+      "Install",
+      "Abort"
+    );
+    if (response === "Abort") {
+      vscode.window.showErrorMessage(
+        "Unable to build because of missing wasm32-unknown-unknown target."
+      );
+      return;
     }
+    await installWasmTarget();
   }
+  const cwd = path.dirname(editor.document.uri.fsPath);
+  const workspaceFolder = vscode.workspace.getWorkspaceFolder(
+    editor.document.uri
+  );
+  const schemaArgs =
+    schemaSettings === "skip"
+      ? []
+      : schemaSettings === "embed"
+      ? ["--schema-embed"]
+      : ["--schema-base64-out", "-"];
+  const additionalArgs = config.getAdditionalBuildArgs();
+  const args = schemaArgs.concat(additionalArgs);
+  return vscode.tasks.executeTask(
+    await cargoConcordium.build(cwd, workspaceFolder, args)
```

Comment:
I feel like we should provide some sort of feedback on success.

Like tell them "your module is built in ...". If I see correctly we don't set any output path, so their module is available in an odd location. It would be better to allow them to select this.

Reply:
They will see a panel with the output from cargo-concordium, so they are provided feedback.

Regarding the odd location of the output. I have considered this issue as well, but not sure that I would like to prompt the user for the location every time they run the command.
The solution I went with was having them put it as an additional argument, since they could define a workspace config with this, this is not ideal either.

Reply:
I think it would be better to have some default in the root of the package. Named after the package.

That's at least what I usually do when building contracts and it is very easily discoverable.

Reply:
So now it will output schema and module to an `out` directory. Currently, the resulting module is just named `module.wasm.v1`.